model/cpp/register_model.h

```diff
@@ -168,27 +163,24 @@ class BitFieldModel
     return *this = *this - 1U;
   }
 
-  uint32_t operator--(int)
+  SizeType operator--(int)
   {
-    const uint32_t result = *this;
+    const SizeType result = *this;
     --*this;
     return result;
   }
 
 private:
-  uint32_t m_value;
-
-  static_assert(position >= 0 && position <= 31, "BitFiled position must be between 0 and 31");
-  static_assert(width >= 1 && width <= 32, "BitFiled width must be between 1 and 32");
+  SizeType m_value;
 };
 
-template<uint32_t position>
-class BitFieldModel<position, 1U>
+template<typename SizeType, SizeType position, SizeType width>
+class BitFieldModel<SizeType, position, width, std::enable_if_t<width == 0>>
```

Comment:
Pretty sure it should be `std::enable_if_t<width == 1>`

Reply:
Thanks, you are right.